diff --git a/profiles/watershednow/modules/contrib/feeds/feeds.install b/profiles/watershednow/modules/contrib/feeds/feeds.install
index 7911884..98832ef 100644
--- a/profiles/watershednow/modules/contrib/feeds/feeds.install
+++ b/profiles/watershednow/modules/contrib/feeds/feeds.install
@@ -492,3 +492,28 @@ function feeds_update_6009() {
   db_create_table($ret, 'feeds_push_subscriptions', $table);
   return $ret;
 }
+
+/**
+ * Drop primary key, add auto-increment field as new primary key.
+ * Required for views integration: http://drupal.org/node/654466.
+ */
+
+function feeds_update_6010() {
+  $ret = array();
+
+  db_drop_primary_key($ret, 'feeds_source');
+
+  // Add an auto-increment column and a primary key.
+  $spec = array(
+    'type' => 'serial',
+    'unsigned' => TRUE,
+    'not null' => TRUE,
+    'description' => 'Feed source unique id.',
+  );
+  $new_key = array(
+    'primary key' => array('fid'),
+  );
+  db_add_field($ret, 'feeds_source', 'fid', $spec, $new_key);
+
+  return $ret;
+}
\ No newline at end of file
diff --git a/profiles/watershednow/modules/contrib/feeds/feeds.module b/profiles/watershednow/modules/contrib/feeds/feeds.module
index 515688b..839e4a8 100644
--- a/profiles/watershednow/modules/contrib/feeds/feeds.module
+++ b/profiles/watershednow/modules/contrib/feeds/feeds.module
@@ -244,7 +244,7 @@ function feeds_feeds_plugins() {
  */
 function feeds_nodeapi(&$node, $op, $form) {
 
-  // $node looses any changes after 'validate' stage (see node_form_validate()).
+  // $node loses any changes after 'validate' stage (see node_form_validate()).
   // Keep a copy of title and feeds array between 'validate' and subsequent
   // stages. This allows for automatically populating the title of the node form
   // and modifying the $form['feeds'] array on node validation just like on the
diff --git a/profiles/watershednow/modules/contrib/feeds/views/feeds.views.inc b/profiles/watershednow/modules/contrib/feeds/views/feeds.views.inc
index 42f5760..16acf84 100644
--- a/profiles/watershednow/modules/contrib/feeds/views/feeds.views.inc
+++ b/profiles/watershednow/modules/contrib/feeds/views/feeds.views.inc
@@ -20,7 +20,7 @@ function feeds_views_data() {
   $data['feeds_source']['table'] = array(
     'group' => 'Feeds source',
     'base' => array(
-      'field' => array('feed_nid', 'id'),
+      'field' => 'fid',
       'title' => 'Feeds source',
       'help' => 'The source information for a feed, for example its URL',
     ),
