diff -crB wysiwyg/editors/js/ckeditor-3.0.js good_wysiwyg/editors/js/ckeditor-3.0.js
*** wysiwyg/editors/js/ckeditor-3.0.js	2010-12-19 22:42:54.000000000 -0500
--- good_wysiwyg/editors/js/ckeditor-3.0.js	2011-01-16 19:54:02.081880800 -0500
***************
*** 183,189 ****
                    // content is supposed to contain the "outerHTML".
                    data.content = data.node.$.parentNode.innerHTML;
                  }
!                 data.node = data.node.$;
                }
                else {
                  data.content = '';
--- 183,189 ----
                    // content is supposed to contain the "outerHTML".
                    data.content = data.node.$.parentNode.innerHTML;
                  }
!                 //data.node = data.node.$;
                }
                else {
                  data.content = '';
